### PR TITLE
flow typed update

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers this updates some flow-typed defs. This causes some react-redux things to complain. Affects 2 components
1. Avatar
1. PathItemAction

as a workaround i made .js.flow files. i think what's happening is flow is getting confused by the connector. likely cause things like lifecycle basically destroy the typing

cc: @jzila for pathitemaction

### CODE REVIEW

File: shared/common-adapters/avatar.js.flow [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// @flow
```
Comment (akalin-keybase) [inline shared/common-adapters/avatar.js.flow]: Maybe include a comment as to why the .flow file is needed?

File: shared/stories/prop-providers.js [hunk 2]
```diff
@@ -41,7 +41,7 @@ const WaitingButton = () => ({
 })
 
 const Avatar = (following: string[] = defaultFollowing, followers: string[] = defaultFollowers) => ({
-  Avatar: (ownProps: _Avatar.OwnProps) => _Avatar.mockOwnToViewProps(ownProps, following, followers, action),
+  Avatar: (ownProps: any) => _Avatar.mockOwnToViewProps(ownProps, following, followers, action),
```
Comment (akalin-keybase) [inline shared/stories/prop-providers.js]: This should be `_Avatar.Props`, right?

File: shared/common-adapters/avatar.js.flow [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import * as React from 'react'
+import type {StylesCrossPlatform, StylesCrossPlatformWithSomeDisallowed} from '../styles'
+
+export type AvatarSize = 128 | 96 | 64 | 48 | 32 | 16
+
+type DisallowedStyles = {
+  borderStyle?: empty,
+}
+
+export type Props = {
```
Comment (akalin-keybase) [inline shared/common-adapters/avatar.js.flow]: It's pretty confusing to have `Props` here actually be `OwnProps` in `avatar.js`. Can you make the names match, i.e. rename `Props` here to `OwnProps`, and have another `Props` type that `mockOwnToViewProps` below returns?

File: shared/stories/prop-providers.js (hunk 2 above)
Comment (chrisnojima) [reply shared/stories/prop-providers.js]: we shouldn't even use this mockOwnToView props thing, that should entirely be in here

File: shared/common-adapters/avatar.js.flow (hunk 3 above)
Comment (chrisnojima) [reply shared/common-adapters/avatar.js.flow]: avatar is connected so inside itself it has OwnProps to signify the props that come externally. From the outside those are actually Props. If we rename it, it should be Props instead. But honestly how this is connected is very messy, if we want to fix this up and remove this facade we should just do that

File: shared/stories/prop-providers.js (hunk 2 above)
Comment (chrisnojima) [reply shared/stories/prop-providers.js]: i switched that to _Avatar.Props and the return value of the mocking as any

**Pull request merged**